lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -758,6 +758,40 @@ OpFoldResult AndOp::fold(ArrayRef<Attribute> constants) {
   return constFoldAssociativeOp(constants, hw::PEO::And);
 }
 
+/// returns a single common operand that all inputs of the operation `op` can
+/// be traced back to, or an empty `Value` if no such operand exists.
+///
+/// for example for `or(a[0], a[1], ..., a[n-1])` this function returns `a`
+/// (assuming the bit-width of `a` is `n`)
+template <typename Op>
+static Value getCommonOperand(Op op) {
+  auto inputs = op.inputs();
+
```

Comment:
I think it is necessary to accept only i1 operation, e.g:
```suggestion
  if (!op.getType().isInteger(1))
    return Value();
  auto inputs = op.inputs();
```